src/Gate.php

```diff
@@ -25,16 +28,25 @@ class Gate
      */
     private $signatureHandler;
 
+    /**
+     * Flag validate payment params before generate PaymentPage URL.
+     *
+     * @var bool
+     */
+    private $validateParams;
+
     /**
      * Gate constructor.
      *
      * @param string $secret Secret key
      * @param string $baseUrl Base URL for concatenate with payment params
+     * @param bool $validateParams Validate payment params before generate PaymentPage URL
      */
-    public function __construct($secret, string $baseUrl = '')
+    public function __construct(string $secret, string $baseUrl = '', bool $validateParams = true)
```

Comment:
I don't think passing validateParams marker to the constructor at this point is a good idea.